### PR TITLE
Remove the scoped property store

### PR DESCRIPTION
**Pull request by benogle**
Use of the scoped property store was complex and confusing. This PR takes a much more straightforward approach to storing and filtering the providers. 

Other things improved:

* Specificity sorting uses real specificity
* Sorting of providers is stable, so providers added later will come later when specificities are equal.

### CODE REVIEW

**Comment by joefitzgerald** [comment]
So does this mean that the selector is no longer used to determine specificity?

**Comment by benogle** [comment]
The selector is still used. Previously, the length of the `selector` string was the 'specificity', but this now actually calculates the specificity.

**Comment by joefitzgerald** [comment]
... and this code is actually defining the specificity? https://github.com/atom/selector-kit/blob/d708058abb1d9a92001752ccb92e4313786c90ed/src/selector.coffee#L110-L125

**Comment by benogle** [comment]
yeah

**Comment by joefitzgerald** [comment]
Isn't `*` a valid selector? Why would it be part of the escape regex? https://github.com/atom-community/autocomplete-plus/blob/b8bb2b0b06188726abf00bd57e33491a3edcd117/lib/scope-helpers.coffee#L3

**Comment by benogle** [comment]
This was lifted from scoped-property-store. See: https://github.com/atom/scoped-property-store/commit/c7f8a06917161f7821dd4f589b8e04c7f1c67a27

**Comment by joefitzgerald** [comment]
Gotcha. LGTM.

File: lib/provider-manager.coffee [hunk 1]
```diff
@@ -19,55 +22,44 @@ class ProviderManager
...
-    # Check Global Blacklist For Match With Selector
-    blacklist = _.chain(providers).map((p) -> p.value.globalBlacklist).filter((p) -> p? and p is true).value()
-    return [] if blacklist? and blacklist.length
-
-    # Determine Blacklisted Providers
-    blacklistedProviders = _.chain(providers).filter((p) -> p.value.blacklisted? and p.value.blacklisted is true).map((p) -> p.value.provider).value()
-
-    # TODO API: Remove this when 1.0 API is removed
-    fuzzyProviderBlacklisted = _.chain(providers).filter((p) -> p.value.providerblacklisted? and p.value.providerblacklisted is 'autocomplete-plus-fuzzyprovider').map((p) -> p.value.provider).value() if @fuzzyProvider?
+    selectorsMatchScopeChain ?= require('./scope-helpers').selectorsMatchScopeChain
```
Comment (joefitzgerald) [inline lib/provider-manager.coffee]: Will this perform OK given it's called pretty much on every keystroke?

Comment (benogle) [inline lib/provider-manager.coffee]: I can move it out. It should be fine as it will check null.

**Comment by joefitzgerald** [comment]
Incidentally, the AppVeyor CI job is flakey and fails approximately 50% of the time.

**Pull request merged**